### PR TITLE
NEXT-12785 - [NEW] Administration: Add shortcuts

### PR DESCRIPTION
**Pull request by NiklasLimberg**
What should be done?

Create a new article on our GitBook instance which explains how to add a new shortcut to a module.

This article should mention:

-The prerequisite, a working plugin (Refer to the plugin base guide)
- Every other prerequisite you figure out during writing the guide (e.g. a subscriber, knowing how to create a service, a controller, etc.)
- A short code example, including an explanation, on how to do it
- Maybe this should be built upon the "Add module" guide. If so, please add this reference as a prerequisite.

Category: Extensions > Plugins > Administration
Are there already guides in the previous documentation for this?

Unfortunately, No. But please have a look at our example guide(e.g. note the PLACEHOLDERs for cross-references, that do not exist yet) and the Writing guidelines.

For more information, ask me, Patrick Stahl.

### CODE REVIEW

File: guides/plugins/plugins/administration/add-shortcuts.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+## Adding Shortcuts
```
Comment (lernhart) [inline guides/plugins/plugins/administration/add-shortcuts.md:1]: Should'nt it be: ?
```suggestion
# Adding Shortcuts

## Overview 
```

File: guides/plugins/plugins/administration/add-shortcuts.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+Shortcuts in the Shopware 6 are defined on a Component basis. This guide will show you how.
```
Comment (lernhart) [inline guides/plugins/plugins/administration/add-shortcuts.md:3]:
```suggestion
Shortcuts in Shopware 6 are defined on a Component basis. This guide will show you how.
```

File: guides/plugins/plugins/administration/add-shortcuts.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+## Prerequisites
+
+All you need for this guide is a running Shopware 6 instance and full access to both the files.
```
Comment (lernhart) [inline guides/plugins/plugins/administration/add-shortcuts.md:7]: [...] to both the files **and what** ?

File: guides/plugins/plugins/administration/add-shortcuts.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+Preferably a registered module and custom component.
```
Comment (lernhart) [inline guides/plugins/plugins/administration/add-shortcuts.md:8]: Is this a continuation of the previous statement ? 
If so, seperate them by comma or add an **and**

File: guides/plugins/plugins/administration/add-shortcuts.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+   
+    methods: {
+        function0() {
+            console.log("function0")
+        },
+        function1() {
+            console.log("function1")
+        }
+    }
```
Comment (lernhart) [inline guides/plugins/plugins/administration/add-shortcuts.md:40]:
```suggestion
    shortcuts: {
        'SYSTEMKEY+S': {
            active() {
                return this.acl.can('product.editor');
            },
            method: 'myEditProductFunction'
        },
        ESCAPE: 'myCancelEditProductFunction'
    },

   
    methods: {
        function0() {
            console.log("myEditProductFunction")
        },
        function1() {
            console.log("myCancelEditProductFunction")
        }
    }
```

File: guides/plugins/plugins/administration/add-shortcuts.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+## Prerequisites
+
+All you need for this guide is a running Shopware 6 instance and full access to both the files.
+Preferably a registered module and custom component.
+Of course you'll have to understand JavaScript, but that's a prerequisite for Shopware as a whole and will not be taught as part of this documentation.
+
+## Configuring the Shortcuts
+
+The following code sample will show you how to register shortcuts in your components.
+Shortcuts can be registered with the `shortcuts` attribute.
```
Comment (lernhart) [inline guides/plugins/plugins/administration/add-shortcuts.md:14]:
```suggestion
The following code sample will show you how to register shortcuts in your components with help of the `shortcuts` attribute.
```

File: guides/plugins/plugins/administration/add-shortcuts.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+        },
+        function1() {
+            console.log("function1")
+        }
+    }
+});
+```
+{% endcode %}
+
+The first shortcut is only active if the User has the privilege `product.editor`, learn more about ACL and how to add your own ACL rules [here](./add-acl-rules.md).
```
Comment (lernhart) [inline guides/plugins/plugins/administration/add-shortcuts.md:45]:
```suggestion
The first shortcut is only active if the user has the privilege `product.editor`, learn more about ACL and how to add your own ACL rules [here](./add-acl-rules.md).
```

**Review by lernhart — CHANGES_REQUESTED** [review]

File: guides/plugins/plugins/administration/add-shortcuts.md [hunk 8]
```diff
@@ -0,0 +1,56 @@
+# Adding Shortcuts
+
+## Overview 
+
+Shortcuts in Shopware 6 are defined on a Component basis. This guide will show you how.
```
Comment (leichteckig) [inline guides/plugins/plugins/administration/add-shortcuts.md:5]:
```suggestion
Shortcuts in Shopware 6 are defined on a Component basis. This guide will show you how to add your own ones.
```

**Review by leichteckig — CHANGES_REQUESTED** [review]

File: guides/plugins/plugins/administration/add-shortcuts.md [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+
+The following code sample will show you how to register shortcuts in your components with help of the `shortcuts` attribute.
+
+{% code title="<plugin root>/src/Resources/app/administration/src/module/swag-example/index.js" %}
+```javascript
+const { Component } = Shopware;
+
+Component.register('swag-basic-example', {
+    
+    shortcuts: {
```
[the remark this answers is not in the document]
Comment (NiklasLimberg) [reply guides/plugins/plugins/administration/add-shortcuts.md:23]: i think that  is pretty  self explanatory with property names and  the sentence that is below

File: guides/plugins/plugins/administration/add-shortcuts.md [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+
+## Next steps
+
+Now you that know how to add shortcuts to your components, you may want to learn more about [ACL](./add-acl-rules.md) or try the following things:
+
+* If you want to learn about other possibilities to extend the Administration, we got you covered with other guides:
+  * [Writing templates](writing-templates.md)
+  * [Add custom module](add-custom-module.md)
+  * [Add menu entry](./add-menu-entry.md)
+  * [Add custom routes](./add-custom-route.md)
```
Comment (NiklasLimberg) [inline guides/plugins/plugins/administration/add-shortcuts.md:56]:
```suggestion
* [Writing templates](writing-templates.md)
* [Add custom module](add-custom-module.md)
* [Add menu entry](./add-menu-entry.md)
* [Add custom routes](./add-custom-route.md)
```

File: guides/plugins/plugins/administration/add-shortcuts.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (leichteckig) [reply guides/plugins/plugins/administration/add-shortcuts.md:23]: Don't assume things, this was a great criticism of our old docs. People use the docs to familiarize with shopware so they won't know Shopware's way of shortcuts.

**Review by leichteckig — CHANGES_REQUESTED** [review]

**Review by leichteckig — APPROVED** [review]

**Review by lernhart — APPROVED** [review]

**Review by raknison — APPROVED** [review]

**Review by jleifeld — APPROVED** [review]

File: guides/plugins/plugins/administration/add-shortcuts.md [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+        },
+        myCancelEditProductFunction() {
+            console.log("myCancelEditProductFunction")
+        }
+    }
+});
+```
+{% endcode %}
+
+The first keyboard shortcut reacts to the key combination of `SYSTEMKEY+S`, only if the the user has the privilege `product.editor`, with the invocation of the component method with the name `myEditProductFunction`.
```
Comment (JoshuaBehrens) [inline guides/plugins/plugins/administration/add-shortcuts.md:45]: How can I know what `SYSTEMKEY` means? Is there a list like `SYSTEMKEY: Windows Ctrl / Apple Cmd` or something like that? What is the key called for Alt/⌥?

**Review by JoshuaBehrens — CHANGES_REQUESTED** [review]

**Review by raknison — APPROVED** [review]

File: guides/plugins/plugins/administration/add-shortcuts.md (hunk 11 above)
Comment (JoshuaBehrens) [reply guides/plugins/plugins/administration/add-shortcuts.md:45]: I mean. It is not a list but it is the description I looked for.

**Review by JoshuaBehrens — APPROVED** [review]

**Pull request merged**